From 42dbfb991dc2dd91af7ae5bbe2ea429dcdf8a011 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Sat, 2 Mar 2024 21:19:25 +0100
Subject: [PATCH 17/29] mfmediaengine: Do not send MF_MEDIA_ENGINE_EVENT_ERROR.

Breaks AVPro (used by VRChat) when a subsequent IMFMediaEngine::GetError call doesn't return an IMFMediaError instance
because the source is already shut down.

This should be fixed properly. In the meantime, this hack works around the problem.
---
 dlls/mfmediaengine/main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/mfmediaengine/main.c b/dlls/mfmediaengine/main.c
index 3e05626f238..82bb124bb95 100644
--- a/dlls/mfmediaengine/main.c
+++ b/dlls/mfmediaengine/main.c
@@ -1397,11 +1397,13 @@ static HRESULT WINAPI media_engine_load_handler_Invoke(IMFAsyncCallback *iface,
     }
     else
     {
+#if 0
         engine->network_state = MF_MEDIA_ENGINE_NETWORK_NO_SOURCE;
         engine->error_code = MF_MEDIA_ENGINE_ERR_SRC_NOT_SUPPORTED;
         engine->extended_code = hr;
         IMFMediaEngineNotify_EventNotify(engine->callback, MF_MEDIA_ENGINE_EVENT_ERROR, engine->error_code,
                 engine->extended_code);
+#endif
     }
 
     LeaveCriticalSection(&engine->cs);
-- 
2.44.0

